server/benchmarkmodel/serializers.py

```diff
@@ -114,3 +114,60 @@ def update(self, instance, validated_data):
             instance.approved_at = timezone.now()
         instance.save()
         return instance
+
+
+class ModelPrioritySerializer(serializers.Serializer):
+    priority_rank = serializers.IntegerField(write_only=True)
+    updated_models_order = serializers.JSONField(read_only=True)
+
+    def validate(self, data):
+        if not self.instance:
+            raise serializers.ValidationError("No model association found")
+        if self.instance.approval_status != "APPROVED":
+            raise serializers.ValidationError(
+                "The association should be approved before modifying its priority"
+            )
+        if data["priority_rank"] < -1 or data["priority_rank"] == 0:
+            raise serializers.ValidationError(
+                "The priority value should be a positive integer or -1"
+            )
+        return data
+
+    def update(self, instance, validated_data):
```

Comment:
@johnugeorge Do you suggest having this logic on the client rather than the server?